config/config.go

```diff
@@ -40,5 +43,20 @@ func ReadConfig(cfgFile string, defaultConfig *DefaultConfig, customConfig inter
 		return err
 	}
 
-	return decoder.Decode(c)
+	err = decoder.Decode(c)
+	if err != nil {
+		return err
+	}
+
+	err = validator.Validate(defaultConfig)
```

Comment:
The change needs to be complemented with validation rule configuration on the config struct. Are the validation rules part of another PR?

Reply:
Not sure what your question is. Validation will fail if children are invalid, even if the parent isn't explicitly a `Validator`.

Reply:
Just saw that the validation rules are already annotated on the config structs i.e. validate:"nonnil"`.

Reply:
Yes. Not sure of the history of the validation rules, but they're untouched in this change.